### DISCUSSION TITLE
Change ui.icon element icon name

### DISCUSSION THREAD

**Discussion by thetableman** [zauberzeug/nicegui#1690] [Ideas / Feature Requests]
Other than accessing `_props` is there a way to change the name of an icon referenced in a `ui.icon` element?

It seems amiss that there is no `set_name()` or `bind_name()` function allowing for easy, and consistent, updating of this element. This would be useful when updating icons to reflect warnings, status or, as in my specific case, changing between an up and down arrow.

Would such a feature need a new base class akin to `TextElement`?

**Comment by falkoschindler** [comment]
Yes, you're right, `ui.icon` absolutely deserves a bindable `name` attribute. And I think a `NameElement` mixin similar to `TextElement` (or better: `FilterElement`) is a reasonable approach. At least I don't see a good way to re-use another mixin for this matter.

Would you like to create a pull request? 😀

**Reply by wgong** [reply]
Hi @falkoschindler I suggest to add one sentence for `ui.icon` documentation like: `icon_name` is snake-cased from Material Icons caption name at `Google Fonts` , e.g. `sports_kabaddi` for `Sports Kabaddi`. It is obvious to some, but took me a while to figure this out

**Reply by falkoschindler** [reply]
@wgong That normal usage of material icons. If you select an icon like https://fonts.google.com/icons?icon.set=Material+Icons&selected=Material+Icons+Outlined:account_circle:, you see the code `account_circle` on the right drawer. Furthermore, the demo on https://nicegui.io/documentation/icon shows the code `thumb_up` in snake case.
But sure, the description of the `name` parameter is pretty sparse and could as well be extended mentioning the snake case. 👍

**Reply by falkoschindler** [reply]
Done: https://github.com/zauberzeug/nicegui/commit/da0ec00acf05fe3504ed16178c4457f0c3c6798e
